src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -266,7 +267,19 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
                     {
                         TraceLevel level = (TraceLevel)logData["lvl"];
                         var evt = new TraceEvent(level, message);
-                        traceWriter.Trace(evt);
+
+                        // Node captures the AsyncLocal value of the first invocation, which means that logs
+                        // are correlated incorrectly. Here we'll overwrite that value with the correct value
+                        // immediately before logging.
+                        using (logger.BeginScope(
+                            new Dictionary<string, object>
+                            {
+                                ["MS_FunctionInvocationId"] = invocationContext.ExecutionContext.InvocationId
+                            }))
+                        {
+                            // TraceWriter already logs to ILogger
+                            traceWriter.Trace(evt);
```

Comment:
Not a fan of this - tricky side effects & hard to follow.

I think I'd prefer if we didn't automatically wire up tracewriter in that way for functions - you'd need to add logger duplication but it would be clearer as to what's happening. Also, the AsyncLocal thing might not be an issue in that case? Not to clear on the issue there.